assets/javascripts/discourse/services/chat.js

```diff
@@ -484,25 +470,21 @@ export default Service.extend({
 
     this.setActiveChannel(channel);
 
-    if (
-      Site.currentProp("mobileView") ||
-      this.chatWindowStore.fullPage ||
-      this.fullScreenChatOpen
-    ) {
-      const queryParams = messageId ? { messageId } : {};
-
-      return this.router.transitionTo(
-        "chat.channel",
-        channel.id,
-        slugifyChannel(channel.title),
-        {
-          queryParams,
-        }
-      );
-    } else {
-      this._fireOpenFloatAppEvent(channel, messageId);
-      return Promise.resolve();
-    }
+    next(() => {
```

Comment:
1. Is `next()` necessary here?
2. It's addition made the return statements below no op

Reply:
On second thought I think it's not necessary... I think what I was trying to avoid with this was actually due to something else, will remove :+1: